core/trino-main/src/main/java/io/trino/server/protocol/spooling/CoordinatorSegmentResource.java

```diff
@@ -113,10 +113,14 @@ public static UriBuilder spooledSegmentUriBuilder(ExternalUriInfo info)
                 .path(CoordinatorSegmentResource.class);
     }
 
-    public HostAddress nextActiveNode()
+    public HostAddress randomActiveWorkerNode()
     {
-        List<InternalNode> internalNodes = ImmutableList.copyOf(nodeManager.getActiveNodesSnapshot().getAllNodes());
-        verify(!internalNodes.isEmpty(), "No active nodes available");
+        List<InternalNode> internalNodes = nodeManager.getActiveNodesSnapshot().getAllNodes()
+                .stream()
+                .filter(node -> !node.isCoordinator())
+                .collect(toImmutableList());
+
+        verify(!internalNodes.isEmpty(), "No active worker nodes available");
```

Comment:
this should not be `verify` which is to validate code enforced preconditions. 
You can easily have a cluster without workers if all are shut down.

Reply:
so checkState?

Reply:
It should probably be something which traverses to user nicely. not internal error.
I thinks some TrinoException would be nice